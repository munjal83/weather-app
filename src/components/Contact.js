import React from 'react';
import FormControl from '@material-ui/core/FormControl';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';


const styles = {
    container: {
        display: 'flex',
        flexWrap: 'wrap',
        width: '100%',
        maxWidth: 800,
        backgroundColor: "fff",
        margin: '0 auto',
        padding: 20,
    },
    formControl: {
        margin: 20,
    },
    button: {
        margin: 10,
    },
};

class Contact extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            name: '',
            phone: '',
            email: '',
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        const target = event.target;
        const value = target.value;
        const name = target.name;
        this.setState({ [name]: value });
    }

    handleSubmit(event) {
        event.preventDefault();
    }

    render() {
        return (
            <div style={styles.container}>
                <Paper elevation={3}>
                    <Typography variant="display1" component="h3" align="center" gutterBottom>
                        Contact Us
                    </Typography>
                    <FormControl style={styles.formControl}>
                        <InputLabel htmlFor="name-simple">Name</InputLabel>
                        <Input name="name" value={this.state.name} onChange={this.handleChange} />
                    </FormControl>
                    <FormControl style={styles.formControl}>
                        <InputLabel htmlFor="phone-simple">Phone Number</InputLabel>
                        <Input name="phone" value={this.state.phone} onChange={this.handleChange} />
                    </FormControl>
                    <FormControl style={styles.formControl}>
                        <InputLabel htmlFor="email-simple">Email</InputLabel>
                        <Input name="email" value={this.state.email} onChange={this.handleChange} />
                    </FormControl>
                    <Button
                        variant="contained"
                        color="primary"
                        style={styles.button}
                        href="mailto:admin@example.com"
                    >
                        Contact Us
                    </Button>
                </Paper>
            </div>
        );
    }
}

export default Contact;