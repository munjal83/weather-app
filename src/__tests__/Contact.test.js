import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import Contact from '../components/Contact';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Contact />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Contact Component Snapshot', () => { 
  test('testing contact component', () => {
  let tree = create(<Contact />)
  expect(tree.toJSON()).toMatchSnapshot();
})
});

