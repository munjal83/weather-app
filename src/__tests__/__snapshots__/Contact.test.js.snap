// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component Snapshot testing contact component 1`] = `
<div
  style={
    Object {
      "backgroundColor": "fff",
      "display": "flex",
      "flexWrap": "wrap",
      "margin": "0 auto",
      "maxWidth": 800,
      "padding": 20,
      "width": "100%",
    }
  }
>
  <div
    className="MuiPaper-root-118 MuiPaper-elevation3-123 MuiPaper-rounded-119"
  >
    <h3
      className="MuiTypography-root-145 MuiTypography-display1-149 MuiTypography-gutterBottom-162 MuiTypography-alignCenter-158"
    >
      Contact Us
    </h3>
    <div
      className="MuiFormControl-root-170"
      style={
        Object {
          "margin": 20,
        }
      }
    >
      <label
        className="MuiFormLabel-root-179 MuiInputLabel-root-174 MuiInputLabel-formControl-175 MuiInputLabel-animated-178"
        data-shrink={false}
        htmlFor="name-simple"
      >
        Name
      </label>
      <div
        className="MuiInput-root-186 MuiInput-formControl-187 MuiInput-underline-190"
      >
        <input
          aria-invalid={false}
          className="MuiInput-input-194"
          disabled={false}
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-170"
      style={
        Object {
          "margin": 20,
        }
      }
    >
      <label
        className="MuiFormLabel-root-179 MuiInputLabel-root-174 MuiInputLabel-formControl-175 MuiInputLabel-animated-178"
        data-shrink={false}
        htmlFor="phone-simple"
      >
        Phone Number
      </label>
      <div
        className="MuiInput-root-186 MuiInput-formControl-187 MuiInput-underline-190"
      >
        <input
          aria-invalid={false}
          className="MuiInput-input-194"
          disabled={false}
          name="phone"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-170"
      style={
        Object {
          "margin": 20,
        }
      }
    >
      <label
        className="MuiFormLabel-root-179 MuiInputLabel-root-174 MuiInputLabel-formControl-175 MuiInputLabel-animated-178"
        data-shrink={false}
        htmlFor="email-simple"
      >
        Email
      </label>
      <div
        className="MuiInput-root-186 MuiInput-formControl-187 MuiInput-underline-190"
      >
        <input
          aria-invalid={false}
          className="MuiInput-input-194"
          disabled={false}
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <a
      className="MuiButtonBase-root-225 MuiButton-root-199 MuiButton-contained-210 MuiButton-containedPrimary-211 MuiButton-raised-213 MuiButton-raisedPrimary-214"
      href="mailto:admin@example.com"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={
        Object {
          "margin": 10,
        }
      }
      tabIndex="0"
    >
      <span
        className="MuiButton-label-200"
      >
        Contact Us
      </span>
      <span
        className="MuiTouchRipple-root-228"
      />
    </a>
  </div>
</div>
`;
